src/test/ui/type-alias-impl-trait/bounds-are-checked-2.stderr

```diff
@@ -1,8 +1,8 @@
 error[E0277]: the trait bound `T: Clone` is not satisfied
-  --> $DIR/bounds-are-checked-2.rs:6:13
+  --> $DIR/bounds-are-checked-2.rs:9:5
    |
-LL | type X<T> = impl Clone;
-   |             ^^^^^^^^^^ the trait `Clone` is not implemented for `T`
+LL |     t
+   |     ^ the trait `Clone` is not implemented for `T`
```

Comment:
It's slightly confusing that we're talking about `T` here and `f` also has a `T`, but considering the `help` right below, that should be explanatory enough for this PR I think.

I want to add a new type-alias-impl-trait ObligationCause that we will then use to improve this diagnostic.